internal/resources/result.go

```diff
@@ -31,6 +31,11 @@ func (r Result) Requeue() bool {
 	return r.requeue
 }
 
+// Yield returns true if current processing should be discontinued.
+func (r Result) Yield() bool {
```

Comment:
Now that I see the code I get you idea. Very nice -- indeed, it makes the code more readable.